nbextensions/bulkImportCell/bulkImportCell.js

```diff
@@ -189,34 +223,58 @@ define([
             // widgets this cell owns
             cellTabs,
             controlPanel,
-            fileTypePanel,
-            model = Props.make({
-                data: TestAppObj,
-                onUpdate: function (props) {
-                    Utils.setMeta(this.cell, 'bulkImportCell', props.getRawObject());
-                },
-            });
+            fileTypePanel;
         if (options.initialize) {
-            initialize(typesToFiles);
+            initialize(options.specs);
         }
+        let model = Props.make({
+            data: Utils.getMeta(cell, 'bulkImportCell'),
+            onUpdate: function(props) {
+                Utils.setMeta(cell, 'bulkImportCell', props.getRawObject());
+            }
+        });
 
-        let spec = Spec.make({
-            appSpec: model.getItem('app.spec'),
+        let specs = {};
```

Comment:
There's a little bit of strangeness here. The "specs" that are used by the app cell (and widgets in the BI cell) are somewhat modified from the raw document from the NMS service. Without overwhelming with detail, they get tweaked to know what default values are acceptable, and how they get assorted into "inputs" vs "parameters" vs "outputs".

To keep consistent with the app cell, the "raw" specs are kept around. There's a TODO in the app cell code to modify that, but I wanted to err on the side of consistency for now.